builtin/mainmenu/common.lua

```diff
@@ -43,6 +49,29 @@ function image_column(tooltip, flagname)
 end
 
 --------------------------------------------------------------------------------
+function order_favorite_list(list)
+	local res = {}
+	--orders the favorite list after support
+	for i=1,#list,1 do
+		local fav = list[i]
+		if fav.proto_min == nil or fav.proto_max == nil or not
+			((min_supp_proto > fav.proto_max) or
+			(max_supp_proto < fav.proto_min)) then
```

Comment:
You should add a function instead of writing this test each time, since it is at least used 3 times.